railgun/internal/ctrlutils/ingress-status.go

```diff
@@ -0,0 +1,422 @@
+package ctrlutils
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/kong/deck/file"
+	"github.com/prometheus/common/log"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	knative "knative.dev/networking/pkg/apis/networking/v1alpha1"
+	knativeversioned "knative.dev/networking/pkg/client/clientset/versioned"
+	knativeApis "knative.dev/pkg/apis"
+	"knative.dev/pkg/network"
+
+	"github.com/kong/kubernetes-ingress-controller/pkg/sendconfig"
+	"github.com/kong/kubernetes-ingress-controller/pkg/util"
+	kicclientset "github.com/kong/kubernetes-ingress-controller/railgun/pkg/clientset"
+)
+
+const (
+	status_update_retry = 3
+)
+
+// dedicated function that process ingress/customer resource status update after configuration is updated within kong.
+func PullConfigUpdate(ctx context.Context, kongConfig sendconfig.Kong, log logr.Logger, kubeConfig *rest.Config, publishService string) {
+	ips, hostname, err := RunningAddresses(ctx, kubeConfig, publishService)
+	if err != nil {
+		log.Error(err, "failed to determine kong proxy external ips/hostnames.")
+		return
+	}
+
+	cli, err := clientset.NewForConfig(kubeConfig)
+	if err != nil {
+		log.Error(err, "failed to create k8s client.")
+		return
+	}
+
+	kiccli, err := kicclientset.NewForConfig(kubeConfig)
+	if err != nil {
+		log.Error(err, "failed to create kong ingress client.")
+		return
+	}
+
+	log.Info("Launching Ingress Status Update Thread.")
+
+	var wg sync.WaitGroup
+	for {
+		select {
+		case updateDone := <-kongConfig.ConfigDone:
+			log.V(4).Info("receive configuration information. Update ingress status %v \n", updateDone)
+			wg.Add(1)
+			go UpdateIngress(ctx, &updateDone, log, cli, kiccli, &wg, ips, hostname, kubeConfig)
+		case <-ctx.Done():
+			log.Info("stop status update channel.")
+			wg.Wait()
+			return
+		}
+	}
+}
+
+// update ingress status according to generated rules and specs
+func UpdateIngress(ctx context.Context, targetContent *file.Content, log logr.Logger, cli *clientset.Clientset,
+	kiccli *kicclientset.Clientset,
+	wg *sync.WaitGroup, ips []string, hostname string,
+	kubeConfig *rest.Config) error {
+	defer wg.Done()
+
+	for _, svc := range targetContent.Services {
+
+		for _, plugin := range svc.Plugins {
+			log.Info("\n service host %s name %s plugin enablement %v\n", *svc.Service.Host, *svc.Service.Name, *svc.Plugins[0].Enabled)
```

Comment:
```suggestion
```
I'd still recommend removing this entirely per https://github.com/Kong/kubernetes-ingress-controller/pull/1451/#discussion_r662589925

If it remains in place, it should probably go to `log.V(5).Info()` given that it's going to log for all plugins on every service, even those that aren't related to Knative configuration.

If you think this should remain, can you elaborate on why you think we need to include this log/what would be unclear without it?

Reply:
This is status update footprint,  this record provides accuracy calculation (the metrics, performance provides the data). 
change it to  log.V(5).Info().